File: GJSBackend/WebService/Controllers/PreviewController.cs
// *******************************************************************
// 業務名称　: 互助防疫システム
// 機能概要　: プレビュー
//
// 作成日　　: 2024.07.17
// 作成者　　: AIPlus
// 変更履歴　:
// *******************************************************************
using System.Net;
using BusinessService.Jbd.Gjs.Db;

//using Jbd.Gjs.Common;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Primitives;
using WebService.Common.Base.Response;

namespace Jbd.Gjs.WebService
{
    [Authorize]
    [ApiController]
    [Route("api/AFCT/[controller]")]
    public class PreviewController : WebControllerBase
    {
        public PreviewController(IWebHostEnvironment hostingEnvironment, ServiceFactory serviceFactory) : base(hostingEnvironment, serviceFactory)
        {
        }

        [HttpPost]
        [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
        public IActionResult Preview(WebRequestModel webReq)
        {
            //初期処理
            var service = _serviceFactory.GetService(webReq.SERVICE_NAME);
            var method = service.GetType().GetMethod(webReq.METHOD_NAME)!;

            // パラメータを取得
            var bizReq = RequestFactory.GetRequestObject(webReq.BIZ_REQUEST, method.GetParameters()[0].ParameterType);

            //Sessionの設定
            var r = SetSession<DaRequestBase>(bizReq, true);
            //r.sessionid = DaDbLogService.WriteMainLog(r); modify sf

            //サービス実行
            var res = (CmPreviewResponseBase)method.Invoke(service, new[] { bizReq })!;

            //ログを記録 modify sf
            //DaDbLogService.UpdateLog(webReq.SERVICE_NAME, webReq.METHOD_NAME, startTime, r, res, GetIp(), GetOS(), GetBrowser());
            //using (var db = new DaDbContext(r))
            //{
            //    DaDbLogService.UpdateMainLog(db, res.returncode);
            //    DaDbLogService.WriteTusinLog(db, r, res!, string.Empty);
            //}

            Response.Headers.Append(nameof(CmPreviewResponseBase.returncode), new StringValues(((int)res.returncode).ToString()));
            Response.Headers.Append(nameof(CmPreviewResponseBase.message), new StringValues(WebUtility.UrlEncode(res.message)));
            //if (res.returncode == EnumServiceResult.OK)
            //{
            //    return new FileStreamResult(new MemoryStream(res.data), res.contenttype);
            //}

            return Ok();
        }
    }
}

File: backend/JbdGjsWebService/Controllers/PreviewController.vb
' *******************************************************************
' 業務名称　: 互助防疫システム
' 機能概要　: プレビュー
'
' 作成日　　: 2024.07.17
' 作成者　　: 宋
' 変更履歴　:
' *******************************************************************

Namespace JBD.GJS.WebService
    <Authorize>
    <ApiController>
    <Route("api/GJS/[controller]")>
    Public Class PreviewController
        Inherits WebControllerBase
        Public Sub New(hostingEnvironment As Microsoft.AspNetCore.Hosting.IWebHostEnvironment, serviceFactory As ServiceFactory)
            MyBase.New(hostingEnvironment, serviceFactory)
        End Sub

        <HttpPost>
        <ResponseCache(Duration:=0, Location:=ResponseCacheLocation.None, NoStore:=True)>
        Public Function Preview(webReq As WebRequestModel) As IActionResult

            '初期処理
            Dim service = _serviceFactory.GetService(webReq.SERVICE_NAME)
            Dim method = service.GetType().GetMethod(webReq.METHOD_NAME)

            ' パラメータを取得
            Dim bizReq = RequestFactory.GetRequestObject(webReq.BIZ_REQUEST, method.GetParameters()(0).ParameterType)

            'Sessionの設定
            Dim r = SetSession(Of DaRequestBase)(bizReq, True)

            'チェックトークン
            Dim uid = CheckToken(r.token, Program.ReportsDirectory.ToString())
            If String.IsNullOrEmpty(uid) Then Return New DaResponseBase("トークンが正しくありません。")

            'サービス実行
            r.sessionid = DaDbLogService.WriteMainLog(r)
            Dim res = method.Invoke(service, New Object() {bizReq})
            Dim ret As CmPreviewResponseBase = DirectCast(res, CmPreviewResponseBase)

            'ログを記録
            r.Service = webReq.SERVICE_NAME
            r.Method = webReq.METHOD_NAME
            Using db = New JbdGjsService.JBD.GJS.Service.DaDbContext(r)
                DaDbLogService.UpdateMainLog(db, EnumLogStatus.正常終了)
                DaDbLogService.WriteTusinLog(db, r, res, String.Empty)
            End Using

            If ret.RETURN_CODE = EnumServiceResult.OK Then
                'Return New FileStreamResult(New MemoryStream(ret.data), ret.contenttype)
                Return Ok(New With {Key .report = ret.sectionDocument})
            End If

            Return ret
        End Function
    End Class
End Namespace
